app/models/concerns/eventable.rb

```diff
@@ -0,0 +1,14 @@
+module Eventable
+  extend ActiveSupport::Concern
+
+  def local_payload
```

Comment:
I'd prefer a "event"-related name, so if you see in the model you get a hint of where it may be coming from / used for.

`def local_event_payload`?